### ISSUE TITLE
ILGPU 0.8-beta2 OpenCL

### ISSUE THREAD

**Issue by NicolasG3** [m4rs-mt/ILGPU#77]
I made some tests with the last 0.8 version (I built the 0.8.0-beta2 package from github sources) and experienced some issues with OpenCL backend.

Simple use case not functional with float :
```c#
        static void MyKernel(
            Index1 index,               // The global thread index (1D in this case)
            ArrayView<int> dataView,   // A view to a chunk of memory (1D in this case)
            int constant)              // A sample uniform constant
        {
            float localResult2 = ParabolicInterpolationOffset(0.9f, 0f, 1.1f);

            dataView[index] = index + constant + (int)localResult;
            
            float ParabolicInterpolationOffset(float val1, float val2, float val3)
            {
                float offset = -(val3 - val1) / (2 * (val3 + val1 - 2 * val2));
                return offset;
            }
        }
```
Result :
> Une exception non gérée du type 'System.NotSupportedException' s'est produite dans ILGPU.dll
> The type 'Float32' is not supported in the scope of this arithmetic operation
> 
It seems to be a `float` and `double `mismatch in constant parameters.

Another test with variable parameters
```c#
        static void MyKernel(
            Index1 index,               // The global thread index (1D in this case)
            ArrayView<int> dataView,   // A view to a chunk of memory (1D in this case)
            int constant)              // A sample uniform constant
        {
            float[] myArray = new float[3] { 0.9f, 0f, 1.1f };
            float localResult = ParabolicInterpolationOffset(myArray[0], myArray[1], myArray[2]);

            dataView[index] = index + constant + (int)localResult;
            
            float ParabolicInterpolationOffset(float val1, float val2, float val3)
            {
                float offset = -(val3 - val1) / (2 * (val3 + val1 - 2 * val2));
                return offset;
            }
        }
```
Result :
> Kernel loading failed:
> 3:68:25: error: invalid argument type 'float' to unary expression
>         float var24 = (float)( ~ (float)var23);
>                                ^ ~~~~~~~~~~~~

This comes from unary minus operation.
It works if ...
```c#               float offset = -(val3 - val1) / (2 * (val3 + val1 - 2 * val2));```
... is replaced by ...
```c#               float offset = (val3 - val1) / (2 * (val3 + val1 - 2 * val2));```

In other words ...
```
float val ;
...
val = -val;
``` 
... is not supported by OpenCL backend.

**Comment by m4rs-mt** [comment]
@NicolasG3 Thanks a lot for testing the new backend. I am going to investigate this issue 👍
